test/controllers/ATSPAYEDataControllerTest.scala

```diff
@@ -98,9 +98,9 @@ class ATSPAYEDataControllerTest extends UnitSpec with MockitoSugar with WithFake
       "the service returns no data" in new TestController {
 
```

Comment:
The test that is missing here is when there is a combination of a good response and one of the errors (BAD_REQUEST, INTERNAL_SERVER_ERROR and NOT_FOUND). I think if you had this test in, it would have picked up the issues stated above.

Reply:
Added in more tests

Reply:
You have Ok/404 and Ok/503 but what about the inverse? Carefully consider all the test cases.

Reply:
added in more tests